pkg/analyze/archivereader_test.go

```diff
@@ -0,0 +1,373 @@
+package analyze
+
+import (
+	"fmt"
+	"io/fs"
+	"k8s.io/klog/v2"
+	"reflect"
+	"strings"
+	"testing"
+	"testing/fstest"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"encoding/json"
+)
+
+func brokenOpenFunc(name string) (fs.File, error) {
+	return nil, fmt.Errorf("Broken file")
+}
+
+type wrapperFS struct {
+	base     fstest.MapFS
+	openFunc func(name string) (fs.File, error)
```

Comment:
I don't think it brings enough value to validate this much bloat

Reply:
So I should not use this wrapperFS struct and delete it?

Reply:
I don't see the value that testing the openFunc returning error brings, so imo yes, but feel free to explain why if you think otherwise